int8 width
int8 height
float64[] h_walls
float64[] v_walls